src/browserExt/webRequestIntercept.js

```diff
@@ -0,0 +1,128 @@
+/*
+	***** BEGIN LICENSE BLOCK *****
+	
+	Copyright © 2016 Center for History and New Media
+					George Mason University, Fairfax, Virginia, USA
+					http://zotero.org
+	
+	This file is part of Zotero.
+	
+	Zotero is free software: you can redistribute it and/or modify
+	it under the terms of the GNU Affero General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+	
+	Zotero is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU Affero General Public License for more details.
+
+	You should have received a copy of the GNU Affero General Public License
+	along with Zotero.  If not, see <http://www.gnu.org/licenses/>.
+	
+	***** END LICENSE BLOCK *****
+*/
+
+/**
+ * Handles web request interception and header parsing
+ */
+
+(function() {
+
+'use strict';
+
+Zotero.WebRequestIntercept = {
+	listeners: {
+		beforeSendHeaders: [],
+		headersReceived: [],
+		errorOccurred: []
+	},
+	
+	reqIDToReqHeaders: {},
+
+	init: function() {
+		// Ignore XHR
+		let types = ["main_frame", "sub_frame", "stylesheet", "script", "image", "font", "object", "ping", "other"];
```

Comment:
We do for proxy url rewriting/redirection. Maybe not "ping", although I haven't looked at what that actually is. Although the reason why XHR is ignored escapes me now. Talk about useful comments.